cameo/strain_design/core.py

```diff
@@ -0,0 +1,514 @@
+
+    def to_gnomic(self):
+        """
+        If gnomic is available, return a Gnomic representation of the Target.
+
+        """
+        if _gnomic_available:
+            return Accession(identifier=self.id)
+        else:
+            raise SystemError("Gnomic is only compatible with python >= 3 (%i.%i)" %
+                              (sys.version_info.major, sys.version_info.minor))
+
+    def __str__(self):
+        return self.id
+
+
+class FluxModulationTarget(Target):
+    """
+    A Target that changes the flux constraints (knockouts, over-expression or down-regulation).
+
+    See Also
+    --------
+    ReactionModulationTarget, ReactionKnockoutTarget, GeneModulationTarget and GeneKnockoutTarget for implementation.
+
+    """
+    __gnomic_feature_type__ = 'flux'
+
+    def __init__(self, id, value, reference_value):
+        super(FluxModulationTarget, self).__init__(id)
+        self._reference_value = reference_value
+        self._value = value
+
+    def get_model_target(self, model):
+        raise NotImplementedError
+
+    def apply(self, model, time_machine=None):
+        target = self.get_model_target(model)
+
+        if self._value > 0:
+            target.overexpress(self._value, reference_value=self._reference_value, time_machine=time_machine)
+        elif self._value < 0:
+            target.downregulate(self._value, reference_value=self._reference_value, time_machine=time_machine)
+        else:
+            target.knock_out(time_machine=time_machine)
+
+    def __eq__(self, other):
+        if isinstance(other, FluxModulationTarget):
+            return self.id == other.id and self._value == other._value
+        else:
+            return False
+
+    @property
+    def fold_change(self):
+        return (self._value - self._reference_value) / self._reference_value
+
+    def __str__(self):
+        if self._value == 0:
+            return ui.delta() + self.id
+        elif self._value > self._reference_value:
+            return ui.upreg(self.fold_change) + self.id
+        elif self._value < self._reference_value:
+            return ui.downreg(self.fold_change) + self.id
+
+    def to_gnomic(self):
+        accession = Target.to_gnomic(self)
+        feature = Feature(accession=accession, type=self.__gnomic_feature_type__)
+        if self._value == 0:
+            return Del(feature)
+        elif self._value > self._reference_value:
+            over_expression = Feature(accession=accession, type=self.__gnomic_feature_type__,
+                                      variant="over-expression(%f)" % self.fold_change)
+            return Mutation(feature, over_expression)
+        elif self._value < self._reference_value:
+            under_expression = Feature(accession=accession, type=self.__gnomic_feature_type__,
+                                       variant="down-regulation(%f)" % self.fold_change)
+            return Mutation(feature, under_expression)
+
+
+class ReactionCofactorSwapTarget(Target):
```

Comment:
My opinion is that adding and removing variables is slower than changing coefficients. If I want to see what happens if the same reaction had a different cofactor specificity I would write:
reaction.add_metabolites({nad:-1, nadp:1, nadh:-1, napdh1:}) instead of adding and removing reactions. In the other hand I there are situations where we might want to test both reactions with different cofactors with different ratios, etc...

I'll cook something to solve this problem.